tests/test_stream.py

```diff
@@ -302,6 +302,21 @@ def test_stream_skip_rows_with_headers():
         assert stream.read() == [['2', '中国人']]
 
 
+def test_stream_skip_rows_from_the_end():
+    source = 'data/special/skip-rows.csv'
+    with Stream(source, skip_rows=[-2, 1]) as stream:
+        assert stream.read() == [['1', 'english'], ['2', '中国人']]
+
+    with Stream(source, skip_rows=[-1, -2]) as stream:
+        assert stream.read() == [['id', 'name'], ['1', 'english']]
+
+
+def test_stream_skip_rows_no_double_skip():
```

Comment:
Since skip rows runs before skip negative numbers, it's possible that 'skip_rows' will remove the last line, and then 'skip_negative_numbers' wouldn't know which line is the last line...

Reply:
see previous comment

Reply:
yes, you are right, it is happens :(  fixing...